test/EnvelopesApiTests.js

```diff
@@ -0,0 +1,501 @@
+      doc.documentBase64 = base64Doc;
+      doc.name = 'TestFile.pdf';
+      doc.documentId = '1';
+
+      const docs = [];
+      docs.push(doc);
+      envDef.documents = docs;
+
+      // Add a recipient to sign the document
+      const signer = new docusign.Signer();
+      signer.email = USER_NAME;
+      const name = 'Pat Developer';
+      signer.name = name;
+      signer.recipientId = '1';
+
+      // this value represents the client's unique identifier for the signer
+      const clientUserId = '2939';
+      signer.clientUserId = clientUserId;
+
+      const tabs = getSignerTabsDefinition();
+      signer.tabs = tabs;
+
+      // Above causes issue
+      envDef.recipients = new docusign.Recipients();
+      envDef.recipients.signers = [];
+      envDef.recipients.signers.push(signer);
+
+      // make the envelope with "created" (draft) status
+      envDef.status = 'created';
+
+      envelopesApi.createEnvelope(ACCOUNT_ID, { envelopeDefinition: envDef })
+        .then((envelopeSummary) => {
+          const returnUrl = 'http://www.docusign.com/developer-center';
+          const returnUrlRequest = new docusign.ReturnUrlRequest();
+          returnUrlRequest.returnUrl = returnUrl;
+
+          return envelopesApi.createSenderView(ACCOUNT_ID, envelopeSummary.envelopeId, { returnUrlRequest });
+        })
+        .then((viewUrl) => {
+          if (viewUrl) {
+            console.log(`ViewUrl is ${JSON.stringify(viewUrl)}`);
+            done();
+          }
+        })
+        .catch((error) => {
+          if (error) {
+            return done(error);
+          }
+        });
+    });
+    //
+    it('Update recipients', (done) => {
+      const newSigner = new docusign.Signer();
+      newSigner.email = USER_NAME;
+      newSigner.name = 'Signer2';
+      newSigner.recipientId = '2';
+      //
+      // this value represents the client's unique identifier for the signer
+      const clientUserId = '2939';
+      newSigner.clientUserId = clientUserId;
+      //
+      const tabs = getSignerTabsDefinition();
+      newSigner.tabs = tabs;
+      //
+      const newRecipients = new docusign.Recipients();
+      newRecipients.signers = [];
+      newRecipients.signers.push(newSigner);
+      //
+      envelopesApi.updateRecipients(ACCOUNT_ID, ENVELOPE_ID, { recipients: newRecipients })
+        .then((recipientsUpdateSummary) => {
+          assert.notStrictEqual(recipientsUpdateSummary, undefined);
+          assert.notStrictEqual(recipientsUpdateSummary.recipientUpdateResults, undefined);
+          //
+          return envelopesApi.listRecipients(ACCOUNT_ID, ENVELOPE_ID);
+        })
+        .then((recipients) => {
+          assert.notStrictEqual(recipients, undefined);
+          assert.equal(recipients.recipientCount, 2);
+          done();
+        })
+        .catch((error) => {
+          if (error) {
+            return done(error);
+          }
+        });
+    });
+    //
+    it('Get audit events', (done) => {
```

Comment:
I will comment this test in details, please apply these comments in all tests:
1. The name is not explicit - it should describe what is tested and what is expected result.  Example: Describe("Envelop API" .... Describe("List Audit Events" ... It("should return Ok response"),  It("should return correct list of events"), etc
2. Ideally, there should be one assert per test. For example, you can create expected list of events and do assert.Equal(envelopeAuditEventResponse.auditEvents, expectedEvents);
3. Usually, not only happy path must be covered with tests, but also negative scenarios, like passing wrong Account_Id and Envelop_Id must throw expected exception. Is it Agreed with DocuSign that only happy path cases covered?

Reply:
1. The test names have been updated.
2. Creating a variable with the expected result can be problematic in this case, because the result models contain a large number of fields that may differ after each new request
3. Negative test cases will be added in the scope of the next task